mysql/infoschema.go

```diff
@@ -375,6 +380,41 @@ func getForeignKeys(conv *internal.Conv, db *sql.DB, table schemaAndName) (forei
 	return foreignKeys, nil
 }
 
+// getIndexes return a list of all indexes for the specified table.
+func getIndexes(conv *internal.Conv, db *sql.DB, table schemaAndName) ([]schema.Index, error) {
+	q := `SELECT DISTINCT INDEX_NAME,COLUMN_NAME,SEQ_IN_INDEX,COLLATION,NON_UNIQUE
+		FROM INFORMATION_SCHEMA.STATISTICS 
+		WHERE TABLE_SCHEMA = ?
+			AND TABLE_NAME = ?
+			AND INDEX_NAME != 'PRIMARY' 
+		ORDER BY INDEX_NAME, SEQ_IN_INDEX;`
+	rows, err := db.Query(q, table.schema, table.name)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	var name, column, sequence, collation, nonUnique string
```

Comment:
The type of collation should be something that handles NULL values because the COLLATION column can be NULL (and is for some sakila tables).
Suggest using database/sql's NullString type. And use the Valid method to check the value when reading it.

We should do a quick audit of other information schema queries to check that when we scan we have the appropriate type, both for mysql and postgres. I'll look into this on Monday.

Reply:
This is now done. Resolving comment.